Automattic/jetpack
# My Jetpack: handle AI is_upgradable

This PR addresses some mishaps on the filters and methods involved in the AI class for My Jetpack card to reflect the status properly


Works with D159278-code

See copy edits: p1724199104422529/1724085510.340989-slack-C02TQF5VAJD

Fixes #39075 

## Proposed changes:
Fix a wee glitch with Date constructor when attempting to show remaining days until count reset.
Use helper data to properly determine when to show the "Upgrade" button.

### Other information:


## Jetpack product discussion
p1724085510340989-slack-C02TQF5VAJD

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
Follow the instructions on D159278-code to setup scenarios on your sandbox. Then test them against this PR.
With tiers disabled, only upgradable cases should show the "Upgrade" button. "View" button should always be there.

Enabling tier plans should get things back to normal (both "Upgrade" and "View" buttons are visible)
